client/src/app/pages/dashboard/components/task-item/task-item-in-progress/task-item-in-progress.component.ts

```diff
@@ -0,0 +1,54 @@
+import { Component, EventEmitter, Input, OnInit, Output } from '@angular/core';
+import { FormControl } from '@angular/forms';
+import { EditTask, Task } from '../../../models/models.tasks';
+import { MatDialog } from '@angular/material/dialog';
+import { EditTaskComponent } from 'src/app/pages/dialogs/edit-task/edit-task.component';
+import { DeleteTaskComponent } from 'src/app/pages/dialogs/delete-task/delete-task.component';
+
+@Component({
+  selector: 'app-task-item-in-progress',
+  templateUrl: './task-item-in-progress.component.html',
+  styleUrls: ['./task-item-in-progress.component.scss'],
+})
+export class TaskItemInProgressComponent implements OnInit {
+  @Input() taskItem!: Task;
+
+  @Output() completeTaskEvent = new EventEmitter<undefined>();
+  @Output() editTaskEvent = new EventEmitter<EditTask>();
+  @Output() deleteTaskEvent = new EventEmitter<void>();
+
+  checked = new FormControl(false);
+
+  constructor(public dialog: MatDialog) {
+    this.checked.valueChanges.subscribe((value) => {
+      if (value)
+        setTimeout(() => {
+          this.completeTaskEvent.emit();
+        }, 300);
```

Comment:
Why do we need this timeout?

Reply:
For better user experience; it will show the checkbox being checked before it removes the task item